python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py

```diff
@@ -21,7 +30,7 @@ def __init__(
         api_key: Optional[str] = None,
         api_version: str = "2023-03-15-preview",
         logger: Optional[Logger] = None,
-        ad_auth=False,
+        ad_auth: Union[bool, str, TokenCredential] = False,
```

Comment:
Note sure what the purpose of this being a boolean is. If set to True, the code points to the error path 

> ValueError: The Azure API key cannot be `None` or empty` when not using Azure Credentials

I think it makes more sense to make it:

```suggestion
        ad_auth: Optional[Union[bool, str, TokenCredential]] = None,
```

Reply:
Hi, my reasoning is that if you are not using AD you MUST use an api_key to authenticate.
So even if the default changes to None, IF it is "None" and no key is provided, we should also change the code path to raise the same error.
There MUST be either a Key or the use of AD. 
I left it as "False" just to keep the same default that was already there. But with the appropriate check changes both are ok for me.

Reply:
If it is set to `True` it also needs an `api_key` as an authentication mechanism instead of az credentials, so it is correct to point to the error path.
That is the current behavior.

Reply:
> Hi, my reasoning is that if you are not using AD you MUST use an api_key to authenticate.

By this logic, then wouldn't it make the most sense for ad_auth have to either be false or None to use an api_key? Maybe I'm missing something here, but it looks like you need ad_auth to be a str or TokenCredential to use it for authentication, so for it to ever be 'True' doesn't make sense to me.

Reply:
I'm not exactly sure what is the use case. But that is actually the only case currently supported, and I didn't want to introduce a breaking change.
If you set ad_auth to True, you still need to send an api_key as authentication, but the api_type changes from azure to azure_ad.

Again, I don't know about the logic of that but it is the current behavior.